speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -254,6 +302,14 @@ static void SaveSettingsToDisk()
     // Save black box filepath to xml file
     GfParmSetStr(readParam, PRM_BLACKBOX, GFMNU_ATTR_PATH, m_blackBoxFilePath);
 
+    // Save environment filepath to xml file if a track has been selected
+    // Speed Dreams identifies tracks by their category and their name, so save these two
+    if (m_environmentChosen)
+    {
+        GfParmSetStr(readParam, PRM_ENVIRONMENT_CATEGORY, GFMNU_ATTR_PATH, m_environment->getCategoryId().c_str());
+        GfParmSetStr(readParam, PRM_ENVIRONMENT_NAME, GFMNU_ATTR_PATH, m_environment->getId().c_str());
```

Comment:
Why not put them in `Environment` category with `category` and `name` as keys:

```xml
  <section name="Environment">
    <attstr name="category" val="road"/>
    <attstr name="name" val="test-highway"/>
  </section>
  ```
  
  VS
  
  ```xml
  

  <section name="EnvironmentCategory">
    <attstr name="path" val="road"/>
  </section>

  <section name="EnvironmentName">
    <attstr name="path" val="test_highway"/>
  </section>
  ```